### ISSUE TITLE
'monitor' in lr_scheduler_config can't corporate with trainer parameter 'check_val_every_n_epoch'

### ISSUE THREAD

**Issue by Telephone1024** [PyTorchLightning/pytorch-lightning#10496]
## 🐛 Bug

when I set the trainer parameter `check_val_every_n_epoch=2` while I set `monitor` on 'val/acc_epoch' of my lr_scheduler_config in `configure_optimizers` method, the error occured when the first epoch ended with message `"pytorch_lightning.utilities.exceptions.MisconfigurationException: ReduceLROnPlateau conditioned on metric val/acc_epoch which is not available."`

### To Reproduce
my code is here, I choosed plateau as the sche.sche_type
```python
def configure_optimizers(self):
        optim_config = self.hparams.opt.optim
        params = self.parameters()
        optimizer = torch.optim.AdamW(params, lr=optim_config.lr, eps=optim_config.eps)
        
        if 'plateau' == optim_config.sche.sche_type:
            scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
                optimizer,
                mode='max',
                factor=optim_config.sche.decay_rate,
                patience=optim_config.sche.patience_epochs,
                min_lr=optim_config.min_lr
            )
            lr_scheduler_config = {'scheduler': scheduler, 'monitor': 'val/acc_epoch'}
        elif 'step' == optim_config.sche.sche_type:
            scheduler = torch.optim.lr_scheduler.StepLR(
                optimizer,
                step_size=optim_config.sche.decay_epochs
            )
            lr_scheduler_config = {'scheduler': scheduler, 'interval': 'epoch'}
        
        return {'optimizer': optimizer, 'lr_scheduler': lr_scheduler_config}
```

while I set `check_val_every_n_epoch=2` in trainer

### Expected behavior

I expect that the program keeps running with ReduceLROnPlateau if `check_val_every_n_epoch` is not equal to 1

### Environment
- PyTorch Lightning Version (e.g., 1.5.0):
- PyTorch Version (e.g., 1.10)
- Python version:
- OS (e.g., Linux): Ubuntu 18.04
- CUDA/cuDNN version: cuda:102/cudnn: 7.6.5

**Comment by rohitgr7** [comment]
you need to set the `frequency=check_val_every_n_epoch` accordingly. This is already documented [here](https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers) (check 2nd example) 

The reason we don't check this internally because we can't conclude whether the monitor value set inside scheduler configu belongs to train or val.

**Comment by Telephone1024** [comment]
> you need to set the `frequency=check_val_every_n_epoch` accordingly. This is already documented [here](https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers) (check 2nd example)
> 
> The reason we don't check this internally because we can't conclude whether the monitor value set inside scheduler configu belongs to train or val.

thanks a lot
